Source/WebKit/WebProcess/WebPage/ios/WebPageIOS.mm

```diff
@@ -2477,13 +2477,20 @@ static inline bool rectIsTooBigForSelection(const IntRect& blockRect, const Loca
     constexpr OptionSet hitType { HitTestRequest::Type::ReadOnly, HitTestRequest::Type::Active, HitTestRequest::Type::AllowVisibleChildFrameContentOnly };
     auto* localMainFrame = dynamicDowncast<LocalFrame>(m_page->mainFrame());
     if (!localMainFrame)
-        return;
+        return completionHandler(false, { });
     Ref frame = *localMainFrame;
     auto result = frame->eventHandler().hitTestResultAtPoint(point, hitType);
     RefPtr hitNode = result.innerNonSharedNode();
     if (!hitNode)
         return completionHandler(false, { });
 
+    if (auto* view = frame->view()) {
```

Comment:
```suggestion
    if (RefPtr view = frame->view()) {
```

Reply:
@whsieh why is this better?

Reply:
Generally speaking, we should use `Ref` and `RefPtr` when storing references to ref-counted objects on the stack.

More specifically, when using `auto* view = …`, `view` is just a raw pointer on the stack; this might not be an issue if we're only using trivial getters to read state on the frame view, but since `FrameSelection::contains` triggers layout (via hit-testing), there's no guarantee that the `view` pointer is valid through its entire lifetime. You could imagine that if we added code later on that referenced `view` after calling `frame->selection().contains(pointInContents)`, we would have a security bug (UAF).